### PR TITLE
Widgetize Babble

### PR DESCRIPTION
**Pull request by angusmcleod**
I've widgetized the main components of babble to help whomever is working on this.

Will add more notes in the babble topic.

### CODE REVIEW

File: assets/javascripts/discourse/lib/babble.js.es6 [hunk 1]
```diff
@@ -35,7 +35,10 @@ export default Ember.Object.create({
       }
       self.set('currentTopicId', topic.id)
       messageBus.subscribe('/babble/topics/' + self.get('currentTopicId'), self.setCurrentTopic)
-      messageBus.subscribe('/babble/topics/' + self.get('currentTopicId') + '/posts', self.handleNewPost)
+      messageBus.subscribe('/babble/topics/' + self.get('currentTopicId') + '/posts', (data) => {
+        self.handleNewPost(data)
+        console.log(data)
```
Comment (xfix) [inline assets/javascripts/discourse/lib/babble.js.es6]: Debugging feature was left here.

**Comment by gdpelican** [comment]
This has been merged into the beta branch, thanks for the massive effort. <3

File: assets/javascripts/discourse/initializers/babble-edits.js.es6 [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    withPluginApi('0.1', api => {
+      api.decorateWidget('header-icons:before', function(helper) {
+
+        var currentUser = api.getCurrentUser(),
+            headerState = helper.widget.parentWidget.state,
+            enabled = Discourse.SiteSettings.babble_enabled,
+            topicId = Discourse.Babble.currentTopicId;
+
+        var contents = [];
+        if (!helper.widget.site.mobileView && currentUser && enabled && topicId) {
```
Comment (gdpelican) [inline assets/javascripts/discourse/initializers/babble-edits.js.es6]: Hey @angusmcleod what's the reasoning behind `helper.widget.site.mobileView` here?

Comment (angusmcleod) [inline assets/javascripts/discourse/initializers/babble-edits.js.es6]: @gdpelican Hey. Basically, I hadn't checked the functionality on mobile at all, so I precluded it as a precaution.

**Pull request closed without merging**